Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx

```diff
@@ -0,0 +1,261 @@
+import { FlagIcon } from '@thewca/wca-components'
+import React, { useEffect, useMemo, useState } from 'react'
+import { Link, useParams } from 'react-router-dom'
+import { Checkbox, Popup, Table } from 'semantic-ui-react'
+import { getAllRegistrations } from '../../../api/registration/get/get_registrations'
+import getCompetitorInfo from '../../../api/user/get/get_user_info'
+import LoadingMessage from '../../../ui/loadingMessage'
+import styles from './list.module.scss'
+import RegistrationActions from './RegistrationActions'
+
+const partitionRegistrations = (registrations) => {
+  return registrations.reduce(
+    (result, registration) => {
+      switch (registration.registration_status) {
+        case 'waiting':
+          result.waiting.push(registration)
+          break
+        case 'accepted':
+          result.accepted.push(registration)
+          break
+        case 'deleted':
+          result.deleted.push(registration)
+          break
+        default:
+          break
+      }
+      return result
+    },
+    { waiting: [], accepted: [], deleted: [] }
+  )
+}
+
+export default function RegistrationAdministrationList() {
+  const { competition_id } = useParams()
+  const [registrations, setRegistrations] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [selected, setSelected] = useState({
```

Comment:
Might be worth using a reducer here (via `useReducer`) so that the logic of updating the arrays can be separated into the reducer helper function, and the UI below can look simpler, something like
```
          add={(attendee) => dispatchSelected({ action: 'add-waiting', attendee })
```
or
```
          add={(attendee) => dispatchSelected({ action: 'toggle-waiting', attendee })
```
rather than
```
          add={(attendee) =>
            setSelected({
              waiting: [...selected.waiting, attendee],
              accepted: selected.accepted,
              deleted: selected.deleted,
            })
```

Reply:
TIL about useReducer. Works great!